capstone-express/server.js

```diff
@@ -658,6 +648,60 @@ app.delete("/list/deleteList/:listId", verifyUser, async (req, res) => {
   }
 });
 
+//Delete subject and any children
+async function deleteSubject(subjectId) {
+  const subjectParent = SubjectModel.findById({ _id: subjectId });
+
+  if (subjectParent.list) {
+    for (let i = 0; i < subjectParent.list.length; i++) {
+      await deleteList(subjectParent.list[i]._id);
+    }
+  }
+  await SubjectModel.deleteOne({ _id: subjectId });
+}
+
+app.delete(
+  "/subject/deleteSubject/:subjectId",
+  verifyUser,
+  async (req, res) => {
+    const subjectId = req.params.subjectId;
+
+    try {
+      await deleteSubject(subjectId);
+      res.status(200).json("Subject Deleted");
+    } catch (error) {
+      res.status(404).json(error);
+    }
+  }
+);
+//Delete all user data
+async function deleteAll(userId){
+  const board = await BoardModel.findOne({userId: userId});
+  const subject = await SubjectModel.find({boardId: board._id});
+
+  //Delete all subjects in the board
+  if(subject){
+    for(let i = 0; i < subject.length; i++){
+      await deleteSubject(subject[i]._id);
```

Comment:
This is a inefficient way to delete subjects, since it waits for one action to finish to start the next. You can 
1. store the promises in a array and use Promise.all()
2. Even better use deleteMany()